File: ios/StockPro/Data/DTOs/ReorderRuleDTOs.swift
import Foundation

struct ReorderRuleDTO: Decodable, Identifiable {
    let id: Int
    let product_id: Int
    let product_name: String?
    let supplier_id: Int?
    let supplier_name: String?
    let warehouse_id: Int
    let warehouse_name: String?
    let min_quantity: Int
    let max_quantity: Int
    let trigger_type: String?
    let is_active: Bool
    let created_at: String?
}

struct ReorderRuleCreateRequest: Encodable {
    let product_id: Int
    let supplier_id: Int?
    let warehouse_id: Int
    let min_quantity: Int
    let max_quantity: Int
    let trigger_type: String?
}

struct ReorderRuleUpdateRequest: Encodable {
    let min_quantity: Int?
    let max_quantity: Int?
    let trigger_type: String?
    let is_active: Bool?
    let supplier_id: Int?
}

struct ReplenishmentSuggestion: Decodable, Identifiable {
    let id: Int
    let product_id: Int
    let product_name: String
    let current_stock: Int
    let suggested_quantity: Int
    let warehouse_id: Int
    let warehouse_name: String?
    let supplier_name: String?
}

File: App2/core/data/src/main/java/com/app2/core/data/remote/dto/ReorderRuleDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ReorderRuleDTO(
    val id: Int,
    @SerialName("product_id") val productId: Int,
    @SerialName("product_name") val productName: String? = null,
    @SerialName("warehouse_id") val warehouseId: Int? = null,
    @SerialName("min_quantity") val minQuantity: Int? = null,
    @SerialName("max_quantity") val maxQuantity: Int? = null,
    @SerialName("trigger_type") val triggerType: String? = "manual",
    @SerialName("supplier_id") val supplierId: Int? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class ReorderRuleCreateRequest(
    @SerialName("product_id") val productId: Int,
    @SerialName("warehouse_id") val warehouseId: Int? = null,
    @SerialName("min_quantity") val minQuantity: Int? = null,
    @SerialName("max_quantity") val maxQuantity: Int? = null,
    @SerialName("trigger_type") val triggerType: String? = null,
    @SerialName("supplier_id") val supplierId: Int? = null
)

@Serializable
data class ReplenishmentSuggestion(
    val id: Int? = null,
    @SerialName("product_id") val productId: Int,
    @SerialName("product_name") val productName: String? = null,
    @SerialName("current_quantity") val currentQuantity: Int? = null,
    @SerialName("suggested_quantity") val suggestedQuantity: Int? = null,
    @SerialName("warehouse_id") val warehouseId: Int? = null,
    @SerialName("warehouse_name") val warehouseName: String? = null
)
